### PR TITLE
add ut of ads_loader.go

### PR DESCRIPTION
**Pull request by LiZhenCheng9527**
**What type of PR is this?**
/kind enhancement

**What this PR does / why we need it**:
add ut of ads_loader.go
**Which issue(s) this PR fixes**:
a part of #104

### CODE REVIEW

**Review by hzxuzhonghu — COMMENTED** [review]
LGTM. I think we can also test the dynamicLoader status.

Further more we need to test the xds time sequence

**Comment by LiZhenCheng9527** [comment]
I'll add test cases later.

File: pkg/controller/ads/cache_test.go [hunk 1]
```diff
@@ -0,0 +1,623 @@
+/*
+ * Copyright 2023 The Kmesh Authors.
```
Comment (nlgwcy) [inline pkg/controller/ads/cache_test.go:2]: Copyright 2024 The Kmesh Authors.

**Pull request merged**